backend/packages/Upgrade/src/api/controllers/FeatureFlagController.ts

```diff
@@ -64,6 +135,64 @@ interface FeatureFlagsPaginationInfo extends PaginationResponse {
 export class FeatureFlagsController {
   constructor(public featureFlagService: FeatureFlagService) {}
 
+  /**
+   * @swagger
+   * /flags:
+   *    get:
+   *       description: Get all the feature flags
+   *       tags:
+   *         - Feature Flags
+   *       produces:
+   *         - application/json
+   *       responses:
+   *          '200':
+   *            description: Feature Flag List
+   *            schema:
+   *              type: array
+   *              items:
+   *                $ref: '#/definitions/FeatureFlag'
+   *          '401':
+   *            description: AuthorizationRequiredError
+   */
+
+  @Get()
+  public find(@Req() request: AppRequest): Promise<FeatureFlag[]> {
+    return this.featureFlagService.find(request.logger);
+  }
+
+  /**
+   * @swagger
+   * /flags/{id}:
+   *    get:
+   *       description: Get feature flag by id
+   *       parameters:
+   *         - in: path
+   *           name: id
+   *           required: true
+   *           schema:
+   *             type: string
+   *           description: Feature Flag Id
+   *       tags:
+   *         - Feature Flags
+   *       produces:
+   *         - application/json
+   *       responses:
+   *          '200':
+   *            description: Get Feature Flag By Id
+   *            schema:
+   *                $ref: '#/definitions/FeatureFlag'
+   *          '401':
+   *            description: AuthorizationRequiredError
+   *          '404':
+   *            description: Feature Flag not found
+   *          '500':
```

Comment:
i'd expect this to be a 400 bad request if the id type is wrong

Reply:
I think we that's kind of missing from swagger in a lot of places, not just here (400 = bad request)

Reply:
I agree that's probably a more correct error, but it looks like other places where we expect a UUID in a GET ('/experiments/single/:id', '/segments/status/:segmentId), we throw a 500. I'll follow that for now.